### PR TITLE
Make prometheus support optional

### PR DESCRIPTION
**Pull request by posix4e**

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+0. All of the help wanted issues have a bounty, usually a minimum of 100$ unless they are trivial.
+1. Please ask on #rust-metrics to get the details.
 
 Contributing to rust-metrics
 -----------------------
 
 Please use `cargo fmt` as part of the commit process.
 
-0. Check our issues for things marked ready to do
+0. Check our issues for things marked Ready
```
Comment (waywardmonkeys) [inline Contributing.md:12]: You can link this to the actual search for Ready issues.

File: src/lib.rs [hunk 2]
```diff
@@ -22,15 +22,21 @@
...
+extern crate iron;
+#[cfg(prometheus)]
+extern crate router;
+#[cfg(prometheus)]
+extern crate persistent;
+#[cfg(prometheus)]
 extern crate protobuf; // depend on rust-protobuf runtime
+#[cfg(prometheus)]
 #[allow(unsafe_code)]
+#[cfg(prometheus)]
```
Comment (waywardmonkeys) [inline src/lib.rs:40]: Are both of these `cfg(prometheus)` needed here?

**Comment by waywardmonkeys** [comment]
Some of these commits should be moved out to something separate, I think and rebased on top of master after your last updates?

**Comment by posix4e** [comment]
@waywardmonkeys I rejiggered the commits. That look better to you?

**Pull request merged**